Extension/src/LanguageServer/extension.ts

```diff
@@ -199,6 +222,253 @@ function onInterval(): void {
     clients.ActiveClient.onInterval();
 }
 
+async function parseJsonAtPath(path: string): Promise<any> {
+    try {
+        const exists: boolean = await util.checkFileExists(path);
+        if (exists) {
+            const fileContent: string = await util.readFileText(path);
+            return JSON.parse(fileContent);
+        }
+    } catch (error) {
+    }
+}
+
+interface ParsedVersion {
+    major: number;
+    minor: number;
+    patch: number;
+    suffix?: string;
+}
+
+function getParsedVersion(versionStr: string): ParsedVersion | undefined {
+    let tokens: string[] = versionStr.split(new RegExp('[-\\.]', 'g')); // Match against dots and dashes
+    if (tokens.length < 3) {
+        telemetry.logLanguageServerEvent('versionParsingFailure', { 'versionString': versionStr });
+        return;
+    }
+
+    const parsedVersion: ParsedVersion = function(tokens): ParsedVersion {
+        let parsedVersion: ParsedVersion;
+        parsedVersion.major = parseInt(tokens[0]);
+        parsedVersion.minor = parseInt(tokens[1]);
+        parsedVersion.patch = parseInt(tokens[2]);
+        parsedVersion.suffix = tokens[3];
+        return parsedVersion;
+    }(tokens);
+
+    if (!parsedVersion.major || !parsedVersion.minor || !parsedVersion.patch) {
+        telemetry.logLanguageServerEvent('versionParsingFailure', { 'versionString': versionStr });
+    }
+
+    return parsedVersion;
+}
+
+function parsedVersionGreater(v1: ParsedVersion, v2: ParsedVersion): boolean {
+    // ParsedVersions cannot be compared if either have a suffix that is not 'insiders'
+    if ((v1.suffix && v1.suffix !== 'insiders') || (v2.suffix && v2.suffix !== 'insiders')) {
+        return false;
+    }
+
+    let diff: number = v2.major - v1.major;
+    if (diff) {
+        return diff > 0;
+    } else
+    if (diff = v2.minor - v1.minor) {
+        return diff > 0;
+    } else
+    if (diff = v2.patch - v1.patch) {
+        return diff > 0;
+    } else
+    if (!v2.suffix && v1.suffix === 'insiders') {
+        return true;
+    }
+    return false;
+}
+
+interface Asset {
```

Comment:
I think all of the stuff related to the GitHub API should move to its own file.  Export a single function that gets you a VSIX download url.  Something like `export function getTargetBuild(updateChannel: string): string`.  If an empty string or undefined is returned, you do nothing.

All of the github api stuff, version checking, etc, happens as a result of calling this export.